lib/teleterm/apiserver/handler/handler_clusters.go

```diff
@@ -100,10 +102,11 @@ func newAPIRootCluster(cluster *clusters.Cluster) *api.Cluster {
 
 func newAPILeafCluster(leaf clusters.LeafCluster) *api.Cluster {
 	return &api.Cluster{
-		Name:      leaf.Name,
-		Uri:       leaf.URI.String(),
-		Connected: leaf.Connected,
-		Leaf:      true,
+		Name:       leaf.Name,
+		ActualName: leaf.Name,
```

Comment:
I have confirmed that `leaf.Name` is the correct name for the leaf cluster (the one from the key).